### PR TITLE
Add Coverity scan

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**

### CODE REVIEW

**Comment by artemry-nv** [comment]
Are there any open Coverity issues with high severity (if yes, please open RM bug)? Where can developers see the coverity report in PR?

**Comment by Alexey-Rivkin** [comment]
Test build:
https://dev.azure.com/ucfconsort/xpmem/_build/results?buildId=70685
Coverity report is available as an artifact in the HTML format.

Also both summary and detailed reports are available in the log:
https://dev.azure.com/ucfconsort/xpmem/_build/results?buildId=70685&view=logs&j=03fdef1b-ad3f-5d26-daac-16fc56fa63ed&t=41e24686-70b1-5db5-0fd6-6e2a93ac9df9

Severity rating is unavailable, as the Coverity server is inaccessible from RDMZ.

**Comment by artemry-nv** [comment]
> Test build: https://dev.azure.com/ucfconsort/xpmem/_build/results?buildId=70685 Coverity report is available as an artifact in the HTML format.
> 
> Also both summary and detailed reports are available in the log: https://dev.azure.com/ucfconsort/xpmem/_build/results?buildId=70685&view=logs&j=03fdef1b-ad3f-5d26-daac-16fc56fa63ed&t=41e24686-70b1-5db5-0fd6-6e2a93ac9df9
> 
> Severity rating is unavailable, as the Coverity server is inaccessible from RDMZ.

@tvegas1 could you please check the open coverity issues?

**Review by artemry-nv — DISMISSED** [review]

**Review by tvegas1 — DISMISSED** [review]
will check issues

**Comment by tvegas1** [comment]
@Alexey-Rivkin actually can we modify patch to keep issues that are coming from xpmem dir only? currently some issues are coming from kernel header that we don't own.

**Comment by tvegas1** [comment]
PR fixing xpmem coverity issues: #50

**Comment by Alexey-Rivkin** [comment]
Coverity with updated results:
https://dev.azure.com/ucfconsort/xpmem/_build/results?buildId=70767&view=logs&j=03fdef1b-ad3f-5d26-daac-16fc56fa63ed&t=41e24686-70b1-5db5-0fd6-6e2a93ac9df9

File: ci/coverity.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+      - bash: |
+          set -Exeuo pipefail
+          source /etc/profile.d/modules.sh
+          module load /hpc/local/etc/modulefiles/tools/cov-2021.12
+          ./autogen.sh
+          kernel_ver=$(rpm -qa | grep kernel-devel | cut -d'-' -f3-)
+          ./configure --with-kerneldir=/usr/src/kernels/${kernel_ver}
+          cov-build --dir cov-dir make -j$(nproc) &>/dev/null
+          cov-manage-emit --dir cov-dir --tu-pattern "file('.*/kernel/*')" delete || true
```
Comment (tvegas1) [inline ci/coverity.yml]: seems we are also excluding `xpmem/kernel`. maybe use regexp `/usr/.*/kernel` or first -tu include `xpmem/kernel`?

Comment (Alexey-Rivkin) [reply ci/coverity.yml]: Proper error reporting based on Elena's suggestion:
https://dev.azure.com/ucfconsort/xpmem/_build/results?buildId=70830&view=logs&j=03fdef1b-ad3f-5d26-daac-16fc56fa63ed&t=41e24686-70b1-5db5-0fd6-6e2a93ac9df9

Comment (tvegas1) [reply ci/coverity.yml]: merged #50, do we want to make coverity failure mandatory?

Comment (Alexey-Rivkin) [reply ci/coverity.yml]: It's your call

Comment (tvegas1) [reply ci/coverity.yml]: ok let's make it mandatory

File: ci/azure-pipelines.yml [hunk 2]
```diff
@@ -57,6 +58,10 @@ stages:
             displayName: Check gtest code format
             workingDirectory: $(System.DefaultWorkingDirectory)/test/gtest
 
+  - stage: Coverity
+    jobs:
```
Comment (brminich) [inline ci/azure-pipelines.yml:63]: maybe also make it dependant on code style?

**Review by artemry-nv — DISMISSED** [review]

File: ci/coverity.yml (hunk 1 above)
Comment (Alexey-Rivkin) [reply ci/coverity.yml]: Done

File: ci/azure-pipelines.yml (hunk 2 above)
Comment (Alexey-Rivkin) [reply ci/azure-pipelines.yml:63]: Done

**Review by tvegas1 — APPROVED** [review]

**Pull request merged**